File: srv/src/main/java/customer/ai2code/model/config/AIServiceConfig.java
package customer.ai2code.model.config;

/**
 * 通用AI服务配置接口
 */
public interface AIServiceConfig {
    
    /**
     * 获取AI服务类型
     */
    String getServiceType();
    
    /**
     * 获取API端点URL
     */
    String getApiUrl();
    
    /**
     * 获取认证信息
     */
    AIAuthConfig getAuthConfig();
    
    /**
     * 检查配置是否有效
     */
    boolean isValid();
}

File: srv/config/AIServiceConfig.ts
import { AIAuthConfig } from "./AIAuthConfig";

/**
 * 通用AI服务配置接口
 */
export interface AIServiceConfig {
    /**
     * 获取AI服务类型
     */
    getServiceType(): string;
    
    /**
     * 获取API端点URL
     */
    getApiUrl(): string;
    
    /**
     * 获取认证信息
     */
    getAuthConfig(): AIAuthConfig;
    
    /**
     * 检查配置是否有效
     */
    isValid(): boolean;
}
